File: youpinhui-sellergoods-interface/src/main/java/com/youpinhui/sellergoods/service/GoodsService.java
package com.youpinhui.sellergoods.service;
import java.util.List;

import com.youpinhui.entity.PageResult;
import com.youpinhui.pojo.TbGoods;
import com.youpinhui.pojo.TbItem;
import com.youpinhui.pojogroup.Goods;


/**
 * provider interface
 * @author Administrator
 *
 */
public interface GoodsService {

	/**
	 * find all
	 * @return
	 */
	public List<TbGoods> findAll();
	
	
	/**
	 * query according pagination 
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * Add
	*/
	public void add(Goods goods);
	
	
	/**
	 * update
	 */
	public void update(Goods goods);
	

	/**
	 * get one by id
	 * @param id
	 * @return
	 */
	public Goods findOne(Long id);
	
	
	/**
	 * batch delete
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * fuzzy search 
	 * if the object  is not null , get the parameter from it as the search condition.
	 * @param pageNum current page number
	 * @param pageSize size of the page to display
	 * @return
	 */
	public PageResult findPage(TbGoods goods, int pageNum,int pageSize);
	
	/**
	 * update  status
	 * 
	 */
	public void updateStatus(Long [] ids,String status);
	
	

	/**
	 * update  Marketable
	 * 
	 */
	public void updateMarketable(Long [] ids);
	
	/**
	 * 
	 * search SKU by SPU ids
	 *  
	 **/
	public List<TbItem> searchItemListByGoodsIdListAndStatus(Long[] goodsIds,String status) ;
	
	
}

File: youpinhui-manager-web/src/main/webapp/js/service/goodsService.js
//Sevice
app.service('goodsService',function($http){
	    	
	 //test case , get all the data return a Json
	this.findAll=function(){
		return $http.get('../goods/findAll.do');		
	}
	//pagination
	this.findPage=function(page,rows){
		return $http.get('../goods/findPage.do?page='+page+'&rows='+rows);
	}
	//find one object
	this.findOne=function(id){
		return $http.get('../goods/findOne.do?id='+id);
	}
	//add 
	this.add=function(entity){
		return  $http.post('../goods/add.do',entity );
	}
	//update 
	this.update=function(entity){
		return  $http.post('../goods/update.do',entity );
	}
	//batch delete
	this.dele=function(ids){
		alert(ids);
		return $http.get('../goods/delete.do?ids='+ids);
	}
	//search
	this.search=function(page,rows,searchEntity){
		return $http.post('../goods/search.do?page='+page+"&rows="+rows, searchEntity);
	}    	
	this.updateStatus=function(ids,status){
		return $http.get('../goods/updateStatus.do?ids='+ids+"&status="+status);
	}
});
